tests/meteoroid/src/meteoroid/utilities/vault.py

```diff
@@ -0,0 +1,48 @@
+import hvac
+import os
+import pathlib
+from meteoroid import exceptions
+import logging
+
+
+VAULT_ADDR = "https://clotho.broadinstitute.org:8200"
+VAULT_TOKEN = os.environ['VAULT_TOKEN']
+
+
+def get_authenticated_vault_client(token: str) -> hvac.Client:
+    """Returns an authenticated hvac.Client instance."""
+    if pathlib.Path(token).is_file():
+        logging.info(f"Reading the Vault token from file {token}")
+        with open(token, 'r') as token_file:
+            token = token_file.read()
+
+    client = hvac.Client(url=VAULT_ADDR, token=token)
+
+    if not client.is_authenticated():
+        raise exceptions.VaultAuthException(
+            f"Failed to authenticate with Vault server {VAULT_ADDR}"
+        )
+    return client
+
+
+def read_secret(client: hvac.Client, path: str, pure_data: bool = True) -> dict:
+    """Returns the secret in a dictionary given the path in Vault."""
+    secret = client.read(path=path)
+
+    if not secret:
+        raise exceptions.InvalidVaultSecretPath(
+            f"Failed to retrieve secret from {path}"
+        )
+
+    if pure_data:
```

Comment:
What do you mean by `pure_data`?

Reply:
Vault will return something like:
```JSON
{"metadata1": "blah", "metadata2: "blah", "data": "your_secret"}
```
the `pure_data ` here is meant to get the actual secrets without the metadata from vault. I do think `pure_data` is not a good name. Could you come up with any other var name for this?

Reply:
Hmmm what is the use case for returning the "raw" response from vault? What about always returning the "data" by default like `secret.get('data', None) or secret`?

Reply:
I figured some of the values of the raw response might be helpful to us to debug (e.g. renewable, lease_duration, etc...) But yeah, I think the approach you proposed above seems better!

Reply:
I ended up having `secret.get('data', None)` so it will return `Union[dict, None]` if the secret doesn't have data. This way there's less confusion.